workers/data_refinery_workers/processors/transcriptome_index.py

```diff
@@ -87,6 +87,33 @@ def _prepare_files(job_context: Dict) -> Dict:
     job_context["success"] = True
     return job_context
 
+def _extract_assembly_version(job_context: Dict) -> Dict:
+    """Determine the Ensembl assembly version used for this index.
+
+    Ensembl will periodically release updated versions of the
+    assemblies which are where the input files for this processor
+    comes from.  All divisions other than the main one have identical
+    release versions, but we don't know which division these files
+    came from so we can't just hit thier API again. Therefore, look at
+    the URL we used to get the files because it contains the assembly
+    version.
+
+    I'll admit this isn't the most elegant solution, but since the
+    transcriptome index's only database model is the OriginalFiles
+    until processing is complete, there's no other way to pass this
+    information through to this processor without modifying the
+    OriginalFile model.
+    """
+    original_files = job_context["original_files"]
+
+    for og_file in original_files:
+        if "gtf.gz" in og_file.source_filename:
+            extensionless_url = og_file.source_url[:-7]
```

Comment:
I suppose `-7` here is used because the length of `.gtf.gz` is 7? If it is, it might be better to use `.gtf.gz` on line 110.

Reply:
Is this preferred over something like os.path.basename? What if somehow gtf.gz isn't the final seven characters but happens to be somewhere else in the name (not sure why that would be, but just imagining corner cases).

Reply:
No matter which implementation we use to extract the version number, I think it will be helpful to include the current gtf.gz file's exact format in the docstring of this function: `<species>.<version>.gtf.gz` (in case it may be changed in the future). It will also make the code more readable.

Reply:
`os.path.basename` won't help much here:
```python
>> import os
>>> os.path.basename("/some/really/long/path.with.file.extensions")
'path.with.file.extensions'
```

> What if somehow gtf.gz isn't the final seven characters but happens to be somewhere else in the name (not sure why that would be, but just imagining corner cases).

We don't have many corner cases to worry about, because we're the ones constructing these URLs in `foreman/data_refinery/surveyor/transcriptome_index.py` by formatting our templates. The reason I'm checking for `.gtf.gz` is because there are two OriginalFile records associated with the job and I want to parse out the assembly version from the correct one (the fasta file URL doesn't have the assembly version in it).

In general, this isn't really an approach I'd be happy with (and I even mention so in the docstring), but this is the only way we're storing the assembly version at all in our database and this is a relatively minor issue to rework the transcritome-index model over so close to Keytar Kurt.

However I will make both of the changes Dongbo suggested to improve clarity.